nemo_curator/utils/distributed_utils.py

```diff
@@ -617,54 +617,72 @@ def write_to_disk(
 
     Args:
         df: A Dask DataFrame.
-        output_file_dir: The output file path.
+        output_path: The output file path.
         write_to_filename: Boolean representing whether to write the filename using the "filename" column.
         keep_filename_column: Boolean representing whether to keep or drop the "filename" column, if it exists.
         output_type: The type of output file to write. Can be "jsonl" or "parquet".
 
     """
-    if write_to_filename and "filename" not in df.columns:
+
+    # output_path is a file name
+    if isinstance(output_path, str) and output_path.endswith(".jsonl"):
```

Comment:
@VibhuJawa made a point that `output_path` can end with `.parquet`, the same argument is true for any `.xyz` (including `.jsonl`) since those are legible characters for a directory name, so IMO that's not a strong argument to avoid the check for one extension while not the other.

Reply:
IMO if a user is specifying to write to "filename.jsonl" it's pretty clear that they want to write to a single file, not a directory. I was wondering if there is some type of use case I am missing here?

I think the difference with "filename.parquet" is that Parquet is more flexible here - you can use it for both the single partition and multi partition case. With this PR, the former case will look like:
```
filename.parquet
    part.0.parquet
```
while the latter will look like:
```
filename.parquet
    part.0.parquet
    part.1.parquet
    ...
```
which are both easily readable with `DocumentDataset.read_parquet`. Meanwhile, if you used `filename.jsonl` and wrote to multiple partitions, you wouldn't be able to read it back with `DocumentDataset.read_json` because the file format would look like:
```
filename.jsonl
    0.part
    1.part
    ...
```
And you would get a directory-related error. So IMO the `.jsonl` format should only write out a single partition `.jsonl` file.

Reply:
Thanks for explaining in great detail! This makes sense.